### PR TITLE
Java Benchmark failure

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
Patch fix on the bug @nswamy @piyushghai @andrewfayres 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by piyushghai** [comment]
Looks okay to me. 
Can you also add how you tested it, just so that others might have more context about this fix ?

**Comment by lanking520** [comment]
How to test:
```
bash run_java_inference_bm.sh --model-name ObjectDetection --num-runs 5 --model-path-prefix ../infer/models/resnet50_ssd/resnet50_ssd_model --input-image ../infer/images/dog.jpg --batchsize 5
```
Why does this problem appear?

If you try to bind same cpu() twice, some error can be triggered on the backend.
```
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: Too many slices such that some splits are empty
	at scala.Predef$.require(Predef.scala:224)
	at org.apache.mxnet.ExecutorManager$$anonfun$splitInputSlice$1.apply$mcVI$sp(ExecutorManager.scala:201)
	at org.apache.mxnet.ExecutorManager$$anonfun$splitInputSlice$1.apply(ExecutorManager.scala:198)
	at org.apache.mxnet.ExecutorManager$$anonfun$splitInputSlice$1.apply(ExecutorManager.scala:198)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofInt.foreach(ArrayOps.scala:234)
	at org.apache.mxnet.ExecutorManager$.splitInputSlice(ExecutorManager.scala:198)
	at org.apache.mxnet.module.DataParallelExecutorGroup$$anonfun$decideSlices$3.apply(DataParallelExecutorGroup.scala:348)
	at org.apache.mxnet.module.DataParallelExecutorGroup$$anonfun$decideSlices$3.apply(DataParallelExecutorGroup.scala:338)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.mxnet.module.DataParallelExecutorGroup.decideSlices(DataParallelExecutorGroup.scala:338)
	at org.apache.mxnet.module.DataParallelExecutorGroup.<init>(DataParallelExecutorGroup.scala:312)
	at org.apache.mxnet.module.DataParallelExecutorGroup$Builder.build(DataParallelExecutorGroup.scala:226)
	at org.apache.mxnet.module.Module.bind(Module.scala:288)
	at org.apache.mxnet.infer.Predictor$$anonfun$loadModule$1.apply$mcV$sp(Predictor.scala:210)
	at org.apache.mxnet.infer.Predictor$$anonfun$loadModule$1.apply(Predictor.scala:210)
	at org.apache.mxnet.infer.Predictor$$anonfun$loadModule$1.apply(Predictor.scala:210)
	at org.apache.mxnet.infer.MXNetThreadPoolHandler$$anon$4.call(MXNetHandler.scala:73)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by kalyc** [comment]
Changes look good to me
@mxnet-label-bot add [pr-awaiting-merge]

**Comment by harshp8l** [comment]
@mxnet-label-bot add [pr-awaiting-merge]

File: scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/benchmark/JavaBenchmark.java [hunk 1]
```diff
@@ -102,13 +102,6 @@ public static void main(String[] args) {
                 System.exit(1);
         }
         List<Context> context = getContext();
```
Comment (nswamy) [inline scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/benchmark/JavaBenchmark.java]: can you please not bind in `getContext`?

Comment (nswamy) [reply scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/benchmark/JavaBenchmark.java]: can you rename `getContext` to `bindToDevice`?

**Pull request merged**